cpu/esp32/esp-ieee802154/esp_ieee802154_hal.c

```diff
@@ -0,0 +1,466 @@
+{
+    (void)dev;
+    DEBUG("[esp_ieee802154] %s\n", __func__);
+    return -ENOTSUP;
+}
+
+static int _set_frame_filter_mode(ieee802154_dev_t *dev, ieee802154_filter_mode_t mode)
+{
+    (void)dev;
+    DEBUG("[esp_ieee802154] %s\n", __func__);
+    return -ENOTSUP;
+}
+
+static int _config_addr_filter(ieee802154_dev_t *dev, ieee802154_af_cmd_t cmd, const void *value)
+{
+    (void)dev;
+
+    assert(value);
+
+    const uint8_t *addr = value;
+
+    switch (cmd) {
+    case IEEE802154_AF_SHORT_ADDR:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_AF_SHORT_ADDR %02x:%02x\n",
+              __func__, addr[0], addr[1]);
+        return esp_ieee802154_set_short_address(*((const uint16_t *)value)) ? -EIO : 0;
+    case IEEE802154_AF_EXT_ADDR:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_AF_EXT_ADDR "
+              "%02x:%02x:%02x:%02x:%02x:%02x:%02x:%02x\n", __func__,
+              addr[0], addr[1], addr[2], addr[3], addr[4], addr[5], addr[6], addr[7]);
+        return esp_ieee802154_set_extended_address(addr) ? -EIO : 0;
+    case IEEE802154_AF_PANID:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_AF_PANID %02x:%02x\n",
+              __func__, addr[1], addr[0]);
+        return esp_ieee802154_set_panid(*((const uint16_t *)value)) ? -EIO : 0;
+    case IEEE802154_AF_PAN_COORD:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_AF_PAN_COORD not supported\n", __func__);
+        return -ENOTSUP;
+    }
+
+    return -EINVAL;
+}
+
+static int _config_src_addr_match(ieee802154_dev_t *dev, ieee802154_src_match_t cmd,
+                                  const void *value)
+{
+    (void)dev;
+
+    bool enable = *((bool*)value);
+
+    switch (cmd) {
+    case IEEE802154_SRC_MATCH_EN:
+        DEBUG("[esp_ieee802154] %s: IEEE802154_SRC_MATCH_EN %d\n", __func__, enable);
+        return esp_ieee802154_set_pending_mode(enable
+                                                ? ESP_IEEE802154_AUTO_PENDING_ENABLE
+                                                : ESP_IEEE802154_AUTO_PENDING_DISABLE) ? -EIO : 0;
+    case IEEE802154_SRC_MATCH_SHORT_ADD:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_SRC_MATCH_SHORT_ADD\n", __func__);
+        return esp_ieee802154_add_pending_addr(value, true) ? -ENOMEM : 0;
+    case IEEE802154_SRC_MATCH_SHORT_CLEAR:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_SRC_MATCH_SHORT_CLEAR\n", __func__);
+        return esp_ieee802154_clear_pending_addr(value, true) ? -ENOENT : 0;
+    case IEEE802154_SRC_MATCH_EXT_ADD:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_SRC_MATCH_EXT_ADD\n", __func__);
+        return esp_ieee802154_add_pending_addr(value, false) ? -ENOMEM : 0;
+    case IEEE802154_SRC_MATCH_EXT_CLEAR:
+        DEBUG("[esp_ieee802154] %s: set IEEE802154_SRC_MATCH_EXT_CLEAR\n", __func__);
+        return esp_ieee802154_clear_pending_addr(value, false) ? -ENOENT : 0;
+    }
+
+    return -EINVAL;
+}
+
+void esp_ieee802154_cca_done(bool channel_busy)
```

Comment:
Those appear unused - are they called from within the IDF?

Reply:
Exactly, these are callbacks.

Reply:
Better add a comment that the section is for IDF callbacks